tests/jeos/image_info.pm

```diff
@@ -19,22 +19,97 @@ use version_utils qw(is_sle is_opensuse is_openstack);
+    my $influxdb_user = get_var('_SECRET_INFLUXDB_USER');
+    my $influxdb_pwd = get_var('_SECRET_INFLUXDB_PWD');
+    unless ($influxdb_user || $influxdb_pwd) {
+        record_info('INFO', 'INFLUXDB_SERVER variable is set but "_SECRET_INFLUXDB_USER" and "_SECRET_INFLUXDB_PWD" are missing.');
+        return 0;
+    }
+
+    # CASEDIR var is always set, and if when no specified, the value is "sle" for OSD and "opensuse" for O3
+    #return if (get_required_var('CASEDIR') !~ m/^sle$|^opensuse$/);
+
+    my $openqa_host = get_required_var('OPENQA_HOSTNAME');
+    if ($openqa_host =~ /openqa1-opensuse|openqa.opensuse.org/) {    # O3 hostname
+        $job_url = 'https://openqa.opensuse.org/tests/' . get_current_job_id();
+    }
+    elsif ($openqa_host =~ /openqa.suse.de/) {    # OSD hostname
+        $job_url = 'https://openqa.suse.de/tests/' . get_current_job_id();
+    }
+    return 0 unless ($job_url);
+    # Store credentials in a file to be used by curl commands
+    script_run("read -s influx_creds", 0);
+    type_password("-u $influxdb_user:$influxdb_pwd\n");
+    assert_script_run('echo $influx_creds > influxdb_conf');
+
+    # Check if the data for this image has been already published before.
+    # This will avoid publishing the same data if a job is restarted.
+    my $query = 'curl -K influxdb_conf -i -G "http://' . $influxdb_server . ':8086/query?db=data"';
+    $query .= ' --data-urlencode "q=SELECT \"value\" FROM \"image_size\" WHERE \"image\" = \'' . $image . '\'"';
+    script_run("echo '$query' | tee -a $db_log");
+    my $query_output = script_output("$query 2>&1 | tee -a $db_log", proceed_on_failure => 1);
+
+    # Successful query should return 'HTTP/1.1 200 OK'
+    # Empty records will return  '{"results":[{"statement_id":0}]}'
+    # Existing records will return '{"results":[{"statement_id":0,"series":[{"name":"size","columns":["time","value"],"values"' ...
+    if ($query_output !~ /200 OK/) {
+        record_soft_failure("poo#110221 - There has been a problem with the query.");
+        return 0;
+    } elsif ($query_output =~ /values/) {
+        record_info('DB skip', "Skip pushing the data to the DB. There is already a DB record for this image.");
+        return 0;
+    }
+    record_info('DB ok', "Access is OK and the data will be pushed to the DB.\nquery = $query\nresult = \n$query_output");
+    return 1;
+}
+
+sub push_to_influxdb {
+    my ($self, %args) = @_;
+
+    my $query = 'curl -K influxdb_conf -i -XPOST "http://' . $influxdb_server . ':8086/write?db=data"';
+    $query .= ' --data-binary "' . $args{table};
+    $query .= ',distri=' . get_required_var('DISTRI');
+    $query .= ',version=' . get_required_var('VERSION');
+    $query .= ',arch=' . get_required_var('ARCH');
+    $query .= ',flavor=' . get_required_var('FLAVOR');
+    $query .= ',build=' . get_required_var('BUILD');
+    $query .= ',image=' . $image;
+    $query .= ',job_url=' . $job_url;
+    if ($args{extra}) {
```

Comment:
I think it would be safer to `ref $args{extra} eq 'HASH'`

Reply:
It won't make a difference. This will `undef` if `extra` argument is not set.

```
my %my_hash1;
my %my_hash2;

$my_hash1{extra} = {rpm => 'myrpm'};

if ($my_hash1{extra}) {
    print("1\n");
}
if ($my_hash2{extra}) {
    print("2\n");
}
```
This will print only `1`

Reply:
From `undef` perspective, you are right. However, you wish to iterate over the keys of the hash, therefore you want to ensure you are dealing with a hash.

I also find it codewise descriptive rather than plain `$args{extra}`, which can actually hold any scalar or reference.